### PR TITLE
[MSPAINT] Make imageArea window-less

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Make the canvas border-less. Reduce memory usage.
JIRA issue: [CORE-18867](https://jira.reactos.org/browse/CORE-18867)

## Proposed changes

- Move `imageArea` code into `canvasWindow`.
- Delete `imgarea.cpp`, `imgarea.h`, and `imageArea`.
- Add `CCanvasWindow::ImageToCanvas` and `CCanvasWindow::CanvasToImage` to convert the coordinates.
- Realize drawing of resizing image area.

## TODO

- [x] Do tests.

## Comparison

BEFORE (file size; Debug version):
before

AFTER (file size; Debug version):
after

BEFORE (memory usage; Debug version):
before2

AFTER (memory usage; Debug version):
after2

### CODE REVIEW

File: base/applications/mspaint/canvas.cpp [hunk 1]
```diff
@@ -13,20 +13,79 @@ CCanvasWindow canvasWindow;
...
+    , m_hitSelection(HIT_NONE)
+    , m_whereHit(HIT_NONE)
+    , m_ptOrig { -1, -1 }
 {
+    ::SetRectEmpty(&m_rcNew);
+}
+
+VOID CCanvasWindow::drawZoomFrame(INT mouseX, INT mouseY)
+{
+    // FIXME
```
Comment (JoachimHenze) [inline base/applications/mspaint/canvas.cpp:26]: What specifically does not work here yet?

File: base/applications/mspaint/canvas.cpp [hunk 2]
```diff
@@ -143,60 +249,220 @@ LRESULT CCanvasWindow::OnVScroll(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL&
...
+                CString strCoord;
+                strCoord.Format(_T("%ld, %ld"), pt.x, pt.y);
+                SendMessage(hStatusBar, SB_SETTEXT, 1, (LPARAM) (LPCTSTR) strCoord);
+                break;
+            }
+            default:
+                break;
+        }
+
+        /* rectsel and shape tools always show non-negative numbers when drawing */
```
Comment (JoachimHenze) [inline base/applications/mspaint/canvas.cpp:420]: we could use // within this file.

**Review by JoachimHenze — APPROVED** [review]
Didn't spot any major flaws quickly. Approved therefore. But please wait for a 2nd approval, as the diff is quite large.

File: base/applications/mspaint/canvas.cpp (hunk 2 above)
Comment (katahiromz) [reply base/applications/mspaint/canvas.cpp]: Fixed.

File: base/applications/mspaint/canvas.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/canvas.cpp]: Added comment.

**Pull request merged**